stdlib/filename.ml

```diff
@@ -100,9 +100,7 @@ module Unix : SYSDEPS = struct
     && (String.length n < 2 || String.sub n 0 2 <> "./")
     && (String.length n < 3 || String.sub n 0 3 <> "../")
   let check_suffix name suff =
-    String.length name >= String.length suff &&
-    String.sub name (String.length name - String.length suff)
-                    (String.length suff) = suff
+    String.ends_with ~suffix:suff name
```

Comment:
There's also another `check_suffix` in the Win32 module just below this (line 146) that looks like it could use the same change.

A (very minor) other suggestion is that you could rename the `suff` argument to `suffix` and just have `String.ends_with ~suffix name` as the implementation.

Reply:
When I introduced the `String.ends_with`function I deliberately left the function in the Win32 module like it is since this needs to be case insensitive under windows, i.e. a `String.lowercase_ascii` is applied to the suffices.